nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -568,4 +585,15 @@ public Task fetchTaskById(String documentId) {
         return (Task) getResponse(documentId);
     }
 
+    private void refuseIfDateFound(Object obj) {
+        if (obj instanceof Calendar || obj instanceof Date) {
+            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
```

Comment:
When working with a complex `Map` of properties, it would be better to have the `key` of which value is rejected. The current message is only useful when using the `#set(key, value)` method.